functests/0_config/config.go

```diff
@@ -38,26 +40,41 @@ var _ = Describe("[performance][config] Performance configuration", func() {
 	It("Should successfully deploy the performance profile", func() {
 
 		performanceProfile := testProfile()
-		performanceManifest, found := os.LookupEnv("PERFORMANCE_PROFILE_MANIFEST_OVERRIDE")
-		if found {
+		profileAlreadyExists := false
+
+		if discovery.Enabled() {
 			var err error
-			performanceProfile, err = externalPerformanceProfile(performanceManifest)
-			Expect(err).ToNot(HaveOccurred(), "Failed overriding performance profile", performanceManifest)
-			klog.Warning("Consuming performance profile from ", performanceManifest)
+			performanceProfile, err = profiles.GetByNodeLabels(
```

Comment:
Not sure it's doable, but I'd do another step further, checking for any profile regardless of the machine config pool.
As this is going to be run against an already configured cluster, we don't know what worker pool will it relate to.
In that case, while discovering it, you should also verify that the parameters are compliant with the tests.